aospy/automate.py

```diff
@@ -295,19 +295,28 @@ def _exec_calcs(calcs, parallelize=False, client=None, **compute_kwargs):
         def func(calc):
             """Wrap _compute_or_skip_on_error to require only the calc
             argument"""
-            return _compute_or_skip_on_error(calc, compute_kwargs)
+            return _compute_or_skip_on_error(calc, {'write_to_tar': False})
 
         if client is None:
             n_workers = _n_workers_for_local_cluster(calcs)
             with distributed.LocalCluster(n_workers=n_workers) as cluster:
                 with distributed.Client(cluster) as client:
-                    return _submit_calcs_on_client(calcs, client, func)
+                    result = _submit_calcs_on_client(calcs, client, func)
         else:
-            return _submit_calcs_on_client(calcs, client, func)
+            result = _submit_calcs_on_client(calcs, client, func)
+        _serial_write_to_tar(calcs, **compute_kwargs)
+        return result
     else:
         return [_compute_or_skip_on_error(calc, compute_kwargs)
                 for calc in calcs]
 
+def _serial_write_to_tar(calcs, write_to_tar=True):
+    if write_to_tar:
```

Comment:
I think the logic is more intuitive if this if statement goes before the function call:

```python
write_to_tar = compute_kwargs['write_to_tar']
if write_to_tar:
    _serial_write_to_tar(calcs, write_to_tar=write_to_tar)
```

Or, if you want to keep the if statement where it is, make the function name '_maybe_serial_write_to_tar'

(which supersedes the above comment on L307)